File: app/src/main/java/sgtmelon/scriptum/app/model/item/CursorItem.java
package sgtmelon.scriptum.app.model.item;

/**
 * Модель для сохранения в {@link InputItem} курсора текста
 */
public final class CursorItem {

    private final int valueFrom;
    private final int valueTo;

    public CursorItem(int valueFrom, int valueTo) {
        this.valueFrom = valueFrom;
        this.valueTo = valueTo;
    }

    public int getValueFrom() {
        return valueFrom;
    }

    public int getValueTo() {
        return valueTo;
    }

}

File: app/src/main/java/sgtmelon/scriptum/model/item/CursorItem.kt
package sgtmelon.scriptum.model.item

/**
 * Модель для сохранения в [InputItem] курсора текста
 */
class CursorItem(val valueFrom: Int, val valueTo: Int) {

    fun getValue(undo: Boolean) = if (undo) valueFrom else valueTo

    override fun toString() = "valueFrom = $valueFrom | valueTo = $valueTo"

}